File: WangLulu/Stack/DailyTemperatures.java
package Stack;

import java.util.Deque;
import java.util.LinkedList;

public class DailyTemperatures {
        public int[] dailyTemperatures(int[] temperatures) {
        Deque<Integer> stack = new LinkedList<>();
        int n = temperatures.length;
        int[] result = new int[n];
        for (int i = 0; i < n; i++) {
            while (!stack.isEmpty() && temperatures[i] > temperatures[stack.peek()]) {
                int x = stack.pop();
                result[x] = i - x;
            }
            stack.push(i);
        }
        return result;
    }
}

File: ShengJie Wang/Practice_6/DailyTemperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        int len = temperatures.size();
        vector<int> ans(len, 0);
        stack<int> dt;
        for(int i = 0; i < len; i++){
            while(!dt.empty() && temperatures[i] > temperatures[dt.top()]){
                ans[dt.top()] = i - dt.top();
                dt.pop();
            }
            dt.push(i);
        }
        return ans;
    }
};